datatypes/mcs_int128.h

```diff
@@ -203,6 +180,36 @@ class TSInt128
       return s128Value == static_cast<int128_t>(x);
     }
 
+    inline operator double() const
+    {
+      return toDouble();
+    }
+
+    inline long double toDouble() const
+    {
+      return static_cast<double>(s128Value);
+    }
+
+    inline operator long double() const
+    {
+      return toLongDouble();
+    }
+
+    inline long double toLongDouble() const
+    {
+      return static_cast<long double>(s128Value);
```

Comment:
I think this conversion can overflow, this should be `return static_cast<long double>(TFloat128(s128Value))`.

Reply:
This is a very good question. As you can see I didn't change the behavior so I trusted whatever logic existed before I wrote this. And I believe the logic had been introduced by you in the first place.